src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala

```diff
@@ -24,163 +29,187 @@ case class CromwellRuntimeAttributes(docker: Option[String],
 object CromwellRuntimeAttributes {
   private val log = LoggerFactory.getLogger("RuntimeAttributes")
 
-  def apply(wdlRuntimeAttributes: RuntimeAttributes, backendType: BackendType): CromwellRuntimeAttributes =
-    fromAttributeMap(AttributeMap(wdlRuntimeAttributes.attrs), backendType)
+  def apply(wdlRuntimeAttributes: RuntimeAttributes, backendCall: BackendCall, workflowOptions: Option[WorkflowOptions]): CromwellRuntimeAttributes = {
+    val attributes = for {
+      keys <- validateKeys(wdlRuntimeAttributes, backendCall.backend.backendType)
+      attributesFromTask <- TryUtil.sequenceMap(wdlRuntimeAttributes.evaluate(backendCall.lookupFunction(backendCall.locallyQualifiedInputs), backendCall.engineFunctions))
+      attributesWithDefaults <- Try(getAttributesWithDefaults(attributesFromTask, workflowOptions))
+      validatedAttributes <- validateRuntimeAttributes(attributesWithDefaults)
+    } yield validatedAttributes
 
-  def apply(wdlRuntimeAttributes: RuntimeAttributes, workflowOptions: WorkflowOptions, backendType: BackendType) = {
-    val attributesMap = AttributeMap(wdlRuntimeAttributes.attrs)
 
-    val withDefaults = new AttributeMapTrait {
-      override def keys = attributesMap.keys ++ workflowOptions.getDefaultRuntimeOptionKeys
-      override def unsupportedKeys(bt: BackendType): Seq[String] = attributesMap.unsupportedKeys(bt)
-      override def getSeq(key: RuntimeKey): Option[Seq[String]] = attributesMap.getSeq(key)
-      override def get(key: RuntimeKey): Option[String] = attributesMap.get(key) match {
-        case Some(x) => Some(x)
-        case None => workflowOptions.getDefaultRuntimeOption(key.key).toOption
-      }
+    attributes.get
```

Comment:
yes, an apply should only be returning the actual type (which implies an exception on error)